Framework/DataHandling/test/LoadSampleShapeTest.h

```diff
@@ -0,0 +1,211 @@
+        WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(10, 4);
+    Mantid::API::AnalysisDataService::Instance().add(inputWS, inWs);
+    alg.setPropertyValue("InputWorkspace", inputWS);
+    alg.setPropertyValue("OutputWorkspace", outputWS);
+  }
+
+  void clearWorkspaces(const std::string &inputWS,
+                       const std::string &outputWS) {
+    TS_ASSERT_THROWS_NOTHING(
+        Mantid::API::AnalysisDataService::Instance().remove(inputWS));
+    if (outputWS != inputWS) {
+      TS_ASSERT_THROWS_NOTHING(
+          Mantid::API::AnalysisDataService::Instance().remove(outputWS));
+    }
+  }
+
+  const MeshObject *loadMeshObject(const std::string &inputWS,
+                                   const std::string &outputWS,
+                                   const std::string filename) {
+    LoadSampleShape testLoadSampleShape;
+    testLoadSampleShape.initialize();
+    testLoadSampleShape.setPropertyValue("Filename", filename);
+    prepareWorkspaces(testLoadSampleShape, inputWS, outputWS);
+    TS_ASSERT_THROWS_NOTHING(testLoadSampleShape.execute());
+    TS_ASSERT(testLoadSampleShape.isExecuted());
+    return getMeshObject(outputWS);
+  }
+
+  const MeshObject *getMeshObject(const std::string &outputWS) {
+    MatrixWorkspace_sptr ws = getWorkspace(outputWS);
+    Sample s;
+    TS_ASSERT_THROWS_NOTHING(s = ws->sample());
+    auto &obj = s.getShape();
+    auto mObj = dynamic_cast<const MeshObject *>(&obj);
+    TSM_ASSERT_DIFFERS("Shape is not a mesh object", mObj, nullptr);
+    return mObj;
+  }
+
+  MatrixWorkspace_sptr getWorkspace(const std::string &outputWS) {
+    AnalysisDataServiceImpl &dataStore = AnalysisDataService::Instance();
+    if (dataStore.doesExist(outputWS)) {
+      TS_ASSERT_EQUALS(dataStore.doesExist(outputWS), true);
+      Workspace_sptr output;
+      TS_ASSERT_THROWS_NOTHING(output = dataStore.retrieve(outputWS));
+      MatrixWorkspace_sptr outputWorkspace =
+          boost::dynamic_pointer_cast<MatrixWorkspace>(output);
+      return outputWorkspace;
+    } else {
+      return nullptr;
+    }
+  }
+
+  LoadSampleShape loadShape;
+};
+
+class LoadShapeTestPerformance : public CxxTest::TestSuite {
+public:
+  void setUp() override {
+    auto inWs =
+        WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(10, 4);
+    Mantid::API::AnalysisDataService::Instance().add(inWsName, inWs);
+    for (int i = 0; i < numberOfIterations; ++i) {
+      loadAlgPtrs.emplace_back(setupAlg());
+    }
+  }
+
+  void testLoadSampleShapePerformance() {
+    for (auto alg : loadAlgPtrs) {
+      TS_ASSERT_THROWS_NOTHING(alg->execute());
+    }
+  }
+
+  void tearDown() override {
+    for (int i = 0; i < numberOfIterations; i++) {
+      delete loadAlgPtrs[i];
+      loadAlgPtrs[i] = nullptr;
+    }
+    API::AnalysisDataService::Instance().remove(outWsName);
+    API::AnalysisDataService::Instance().remove(inWsName);
+  }
+
+private:
+  std::vector<LoadSampleShape *> loadAlgPtrs;
```

Comment:
See comment above about just calling execute multiple times. You don't need this.